cylc/flow/run_modes/skip.py

```diff
@@ -0,0 +1,162 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities supporting skip modes
+"""
+from logging import INFO
+from typing import (
+    TYPE_CHECKING, Any, Dict, List, Set, Tuple)
+
+from cylc.flow import LOG
+from cylc.flow.exceptions import WorkflowConfigError
+from cylc.flow.platforms import get_platform
+from cylc.flow.task_outputs import (
+    TASK_OUTPUT_SUBMITTED,
+    TASK_OUTPUT_SUCCEEDED,
+    TASK_OUTPUT_FAILED,
+    TASK_OUTPUT_STARTED
+)
+from cylc.flow.task_state import RunMode
+
+if TYPE_CHECKING:
+    from cylc.flow.taskdef import TaskDef
+    from cylc.flow.task_job_mgr import TaskJobManager
+    from cylc.flow.task_proxy import TaskProxy
+    from typing_extensions import Literal
+
+
+def submit_task_job(
+    task_job_mgr: 'TaskJobManager',
+    itask: 'TaskProxy',
+    rtconfig: Dict[str, Any],
+    workflow: str,
+    now: Tuple[float, str]
+) -> 'Literal[True]':
```

Comment:
If consistency between the different modes is needed, these could be implemented as static methods on classes that inherit from an abstract base class. Otherwise no need to have unused arguments here.

Reply:
I don't think there is right now, though Oliver and I might consider that. But to be honest this is really a copy/paste error.